### PR TITLE
Drag and drop custom fields

### PR DESCRIPTION
**Pull request by yotamberk**
Solves #2863

### CODE REVIEW

File: lib/timeline/component/ItemSet.js [hunk 1]
```diff
@@ -2024,25 +2024,19 @@ ItemSet.prototype._onAddItem = function (event) {
...
+    var newItemData;
     if (event.type == 'drop') {
-      var itemData = JSON.parse(event.dataTransfer.getData("text"))
-      newItemData.content = itemData.content; // content is required 
-      newItemData.type = itemData.type || 'box';
-      newItemData[this.itemsData._fieldId] = itemData.id || util.randomUUID();
-
-      if (itemData.type == 'range' || (itemData.end && itemData.start)) {
-        if (!itemData.end) {
+      var newItemData = JSON.parse(event.dataTransfer.getData("text"))
```
Comment (bradh) [inline lib/timeline/component/ItemSet.js]: Why new `var` here?

File: lib/timeline/component/ItemSet.js [hunk 2]
```diff
@@ -2024,25 +2024,19 @@ ItemSet.prototype._onAddItem = function (event) {
...
+      newItemData.content = newItemData.content ? newItemData.content : 'new item'
+      newItemData.start = newItemData.start ? newItemData.start : (snap ? snap(start, scale, step) : start) 
+      newItemData.type = newItemData.type || 'box';
+      newItemData[this.itemsData._fieldId] = newItemData.id || util.randomUUID();
+
+      if (newItemData.type == 'range' || (newItemData.end && newItemData.start)) {
```
Comment (bradh) [inline lib/timeline/component/ItemSet.js]: Could simplify the conditional case - basically just check for a type of `range` without an `end` date.

File: lib/timeline/component/ItemSet.js (hunk 1 above)
Comment (yotamberk) [inline lib/timeline/component/ItemSet.js]: fixed

File: lib/timeline/component/ItemSet.js (hunk 2 above)
Comment (yotamberk) [inline lib/timeline/component/ItemSet.js]: fixed

**Pull request merged**